src/mon/MDSMonitor.cc

```diff
@@ -1895,7 +1907,8 @@ void MDSMonitor::maybe_replace_gid(mds_gid_t gid,
       info.state != MDSMap::STATE_STANDBY_REPLAY &&
       !pending_fsmap.get_filesystem(fscid)->mds_map.test_flag(CEPH_MDSMAP_DOWN) &&
       (sgid = pending_fsmap.find_replacement_for({fscid, info.rank}, info.name,
-                g_conf->mon_force_standby_active)) != MDS_GID_NONE)
+                g_conf->mon_force_standby_active)) != MDS_GID_NONE
+      && may_replace)
```

Comment:
I think it's better to check may_replace before pending_fsmap.find_replacement_for(). The rest change looks good.